### PR TITLE
updating deps to v0.30.0-beta.0

### PR DESCRIPTION
**Pull request by jaidevmane**
**What type of PR is this?**
Bump k8s.io depedencies to v0.30.0-beta.0, and golang version to 1.22.0 to start releasing 1.30.
We can create 1.30 release branch starting from this commit.
> Uncomment only one, leave it on its own line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release_note:
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module k8s.io/cloud-provider-aws
 
-go 1.21
+go 1.22.0
```
Comment (cartermckinnon) [inline go.mod:3]: I’ve seen a handful of panics in other projects using 1.22, what’s our motivation to update this now?

Comment (jaidevmane) [reply go.mod:3]: Reverting this back to 1.21. We can get out another PR to update it if needed.

Comment (jaidevmane) [reply go.mod:3]: tried reverting to go.1.21, got this message.
```
❯ go mod tidy 
go: k8s.io/apimachinery@v0.30.0-beta.0 requires go >= 1.22.0; switching to go1.22.1
go: downloading go1.22.1 (linux/amd64)
```

Comment (cartermckinnon) [reply go.mod:3]: Gotcha, if k/k has moved to it then we should too 👍

You’ll have to update other go version references as well:

https://github.com/kubernetes/cloud-provider-aws/blob/b0fe7a2e8f86d218c2af5ae553680bfefc06f326/Dockerfile#L17

https://github.com/kubernetes/cloud-provider-aws/blob/b0fe7a2e8f86d218c2af5ae553680bfefc06f326/cloudbuild.yaml#L20

https://github.com/kubernetes/cloud-provider-aws/blob/b0fe7a2e8f86d218c2af5ae553680bfefc06f326/.ko.yaml

https://github.com/kubernetes/cloud-provider-aws/blob/b0fe7a2e8f86d218c2af5ae553680bfefc06f326/.github/workflows/deps.yml#L45

https://github.com/kubernetes/cloud-provider-aws/blob/b0fe7a2e8f86d218c2af5ae553680bfefc06f326/.github/workflows/update-deps.yaml#L22

Comment (jaidevmane) [reply go.mod]: Updated.
References:
https://github.com/kubernetes/kubernetes/commit/70221e8405e8d58273a06fc81554c3fceb717b43

**Pull request merged**